Codegen/src/Nodes/ProgramEnd.cs

```diff
@@ -17,15 +17,26 @@ internal class ProgramEnd: Compiler.Nodes.End, Generated {
         this.OriginalProcName = originalProcName;
 
 
+    }
+    public ProgramEnd(string originalProcName) : base(new Compiler.CodeElements.ProgramEnd())
```

Comment:
Semantic is not good here.
We use this new constructor to create an End Node for a program.
parameter should be called `programName`.